src/YoutubeAtom.tsx

```diff
@@ -164,9 +165,35 @@ export const YoutubeAtom = ({
     const [hasUserLaunchedPlay, setHasUserLaunchedPlay] = useState<boolean>(
         false,
     );
+    const [hasUserHovered, setHasUserHovered] = useState<boolean>(false);
     const player = useRef<YoutubePlayerType>();
 
+    const hasOverlay = overrideImage || posterImage;
+
+    // Show the overlay if:
+    // 1) We have one
+    // AND
+    // 2) it hasn't been clicked upon
+    const showOverlay = hasOverlay && !hasUserLaunchedPlay;
```

Comment:
You can use JSDOC for these, meaning hovering over `showOverlay` anywhere will give you this context.

Reply:
Yes! I love JSDoc, doing this